docs/madr/decisions/053-sni.md

```diff
@@ -0,0 +1,57 @@
+# SNI for MeshService, MeshExternalService etc.
+
+* Status: accepted
+
+## Context and Problem Statement
+
+Currently, the SNI format is `<kuma.io/service>{mesh=<mesh>,tag1=value1}`. For example:
+* `backend{mesh=demo}`
+* `backend{mesh=demo,version=2}`
+
+We need to introduce SNI format for `MeshService` and `MeshExternalService` etc.
+Current format does not work, because we need to differentiate between `kuma.io/service` and MeshService.
+Current format also has other problems:
+* It violates [SNI spec](https://datatracker.ietf.org/doc/html/rfc3546#page-8). It should be a valid hostname. Using characters like `,{/` is not valid.
+  Envoy does not complain about it right now, but it can bite us in the future.
+* It can be too long. See https://github.com/kumahq/kuma/issues/8915
+
+## Considered Options
+
+* New format
+
+## Decision Outcome
+
+Chosen option: "New format", because it solves existing problem and lets us differentiate MeshService from old service.
+
+## Pros and Cons of the Options
+
+### New format
+
+It's important to note that we never parse SNI. We only use it for multizone communication on zone proxies to associate to which service pass the data.
+It means that we can hash the data as long as it's consistent hashing.
+We cannot put tags in the original form because they can violate hostname requirements. For example `/` in `kuma.io/service` is not ok.
+The new format could look like this then:
+`<resource-name>.<mesh-name>.<extra-hash-if-needed>.<resource-type>`
+* `<resource-name>` - name of the resource. Name of MeshService or MeshExternalService
+* `<mesh-name>` - name of the mesh
+* `<extra-hash-if-needed>` - 16 characters of SHA256. In case of MeshService tags it's `SHA256(<tag1>=<value1>,<tag2>=<value2>)` sorted by keys.
+* `<resource-type>` - `meshservice` or `meshexternalservice`
```

Comment:
This sounds good but the order confuses me.

It should least to most specific or vice versa. Now it's a mixed bad.

Reply:
Should we just use less characters for these? (ms, mes, mmzs)

Reply:
I feel like it's better to put hash at the end, so readable information comes first. In some cases, maybe useful for matching.

Reply:
> This sounds good but the order confuses me.
> It should least to most specific or vice versa. Now it's a mixed bad.

Can you elaborate? I don't understand. Resource type sounds good but other place holder confuses you? What do you mean about specificity?

> Should we just use less characters for these? (ms, mes, mmzs)

We can

Reply:
changed resoruce-type to shorter versions